modules/openxoutstreamBidAdapter.js

```diff
@@ -0,0 +1,221 @@
+
+    if (config.getConfig('consentManagement.cmpApi') === 'iab') {
+      defaultParams.x_gdpr_f = 1;
+    }
+  }
+
+  return defaultParams;
+}
+
+function buildOXBannerRequest(bids, bidderRequest) {
+  let queryParams = buildCommonQueryParamsFromBids(bids, bidderRequest);
+  queryParams.aus = utils._map(bids, bid => utils.parseSizesInput(bid.sizes).join(',')).join('|');
+
+  if (bids.some(bid => bid.params.doNotTrack)) {
+    queryParams.ns = 1;
+  }
+
+  if (bids.some(bid => bid.params.coppa)) {
+    queryParams.tfcd = 1;
+  }
+
+  let url = `https://${bids[0].params.delDomain}/v/1.0/avjp`
+  return {
+    method: 'GET',
+    url: url,
+    data: queryParams,
+    payload: {'bids': bids}
+  };
+}
+
+function handleVastResponse(response, serverResponse) {
+  const body = response.body
+  let bidResponses = [];
+  if (response !== undefined && body.vastUrl !== '' && body.pub_rev && body.pub_rev > 0) {
+    let vastQueryParams = parse(body.vastUrl).search || {};
+    const openHtmltag = '<html><head></head><body>';
+    const closeHtmlTag = '</body></html>';
+    const sdkScript = createSdkScript().outerHTML;
+    const placementDiv = createPlacementDiv();
+    placementDiv.dataset.pID = PUBLISHER_ID;
+    const placementDivString = placementDiv.outerHTML;
+    const adResponse = getTemplateAdResponse(body.vastUrl);
+    const adResponseString = JSON.stringify(adResponse);
+    const ymAdsScript = '<script type="text/javascript"> window.__ymAds =' + adResponseString + '</script>';
+
+    let bidResponse = {};
+    bidResponse.requestId = serverResponse.bids[0].bidId;
+    bidResponse.bidderCode = BIDDER_CODE;
+    bidResponse.netRevenue = NET_REVENUE;
+    bidResponse.currency = CURRENCY;
+    bidResponse.cpm = Number(body.pub_rev) / 1000;
+    bidResponse.creativeId = body.adid;
+    bidResponse.height = body.height;
+    bidResponse.width = body.width;
+    bidResponse.vastUrl = body.vastUrl;
+    bidResponse.ttl = TIME_TO_LIVE;
+    bidResponse.mediaType = BANNER;
+    bidResponse.ad = openHtmltag + placementDivString + ymAdsScript + sdkScript + closeHtmlTag;
+
+    response.ph = vastQueryParams.ph;
+    response.colo = vastQueryParams.colo;
+    response.ts = vastQueryParams.ts;
+    bidResponses.push(bidResponse);
+  }
+  return bidResponses;
+}
+registerBidder(spec);
+
+// HELPER FUNCTIONS
+function createSdkScript() {
+  const script = document.createElement('script');
+  script.innerHTML = YM_SCRIPT;
+  return script;
+}
+function createPlacementDiv() {
+  const div = document.createElement('div');
+  div.id = `ym_${PLACEMENT_ID}`;
+  div.classList.add('ym');
+  div.dataset.lfid = `${CR_ID}`;
+  return div
+}
+
+/**
+ * Create a nativeplay template with the placement id and vastURL.
+ * @param vastUrl
+ */
+const getTemplateAdResponse = (vastUrl) => {
+  return {
+    availability_zone: 'us-east-1a',
+    data: [
+      {
+        ads: [
+          {
+            actions: {},
+            ad_id: AD_ID,
+            adv_id: `${AD_ID}`,
```

Comment:
are we sure this is right?

Reply:
Yep - that is accurate.

Reply:
Actually, on second thoughts; @HolzAndrew , can you make `AD_ID` (the constant) a string (because large number JS sadness) - and not populate the `ad_id`key in the JSON? I don't think we use it internally.

Reply:
ok done.